File: WorkBridgeAngular/src/app/wallet/services/payment.service.ts
import { Injectable } from '@angular/core';
import { PaymentStatus } from '../../enums/payment-status.enum';
import { Observable } from 'rxjs';
import { DepositSessionResponseDTO, PaymentResponseDTO } from '../models/payment.model';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../enviroments/environment';

@Injectable({
  providedIn: 'root',
})
export class PaymentService {






  private apiUrl =
    environment.apiUrl + 'payments/';

  constructor(
    private http: HttpClient
  ) { }

  // ==========================================
  // Create Deposit Session
  // ==========================================

  createDeposit(
    userId: number,
    amount: number
  ): Observable<DepositSessionResponseDTO> {

    return this.http.post<DepositSessionResponseDTO>(
      `${this.apiUrl}deposit/${userId}?amount=${amount}`,
      {}
    );

  }

  // ==========================================
  // Get Payment By Id
  // ==========================================

  getById(
    id: number
  ): Observable<PaymentResponseDTO> {

    return this.http.get<PaymentResponseDTO>(
      `${this.apiUrl}${id}`
    );

  }

  // ==========================================
  // Get By Gateway Transaction Id
  // ==========================================

  getByGatewayTransactionId(
    gatewayTransactionId: string
  ): Observable<PaymentResponseDTO> {

    return this.http.get<PaymentResponseDTO>(
      `${this.apiUrl}gateway/${gatewayTransactionId}`
    );

  }

  // ==========================================
  // Get User Payments
  // ==========================================

  getUserPayments(
    userId: number
  ): Observable<PaymentResponseDTO[]> {

    return this.http.get<PaymentResponseDTO[]>(
      `${this.apiUrl}user/${userId}`
    );

  }

  // ==========================================
  // Get All Payments
  // ==========================================

  getAll(): Observable<PaymentResponseDTO[]> {

    return this.http.get<PaymentResponseDTO[]>(
      this.apiUrl
    );

  }

  // ==========================================
  // Get Payments By Status
  // ==========================================

  getByStatus(
    status: PaymentStatus
  ): Observable<PaymentResponseDTO[]> {

    return this.http.get<PaymentResponseDTO[]>(
      `${this.apiUrl}status/${status}`
    );

  }



}

File: project/src/main/java/com/wordbridge/project/payment/PaymentService.java
package com.wordbridge.project.payment;

import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.util.List;

@Service
public interface PaymentService {

    // Create a new deposit payment (PENDING)
    DepositSessionResponseDTO createDeposit(Long userId, BigDecimal amount);


    // Get payment
    PaymentResponseDTO getById(Long id);

    // Find by SSLCommerz transaction id (tran_id)
    PaymentResponseDTO getByGatewayTransactionId(String gatewayTransactionId);

    // User payment history
    List<PaymentResponseDTO> getUserPayments(Long userId);

    // Admin
    List<PaymentResponseDTO> getAll();

    List<PaymentResponseDTO> getByStatus(PaymentStatus status);



    // SSLCommerz callback methods
    void markSuccess(
            String gatewayTransactionId,
            String validationId,
            String paymentMethod
    );

    void markFailed(
            String gatewayTransactionId,
            String reason
    );

    void markCancelled(
            String gatewayTransactionId,
            String reason
    );


}
